federation/pkg/federation-controller/util/test/test_helper.go

```diff
@@ -25,14 +28,110 @@ import (
 	"k8s.io/kubernetes/pkg/client/testing/core"
 	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/golang/glog"
 )
 
+// A structure that distributes eventes to multiple watchers.
+type WatcherDispatcher struct {
+	sync.Mutex
+	watchers    []*watch.FakeWatcher
+	eventsSoFar []*watch.Event
+}
+
+func (wd *WatcherDispatcher) register(watcher *watch.FakeWatcher) {
+	wd.Lock()
+	defer wd.Unlock()
+	wd.watchers = append(wd.watchers, watcher)
+	for _, event := range wd.eventsSoFar {
```

Comment:
This is used for tests so queues can explode :). As of events they are roughly equivalent to updating list. The main problem here is that the events are on goroutines (thus may be sent out of order) to avoid odd locking problems which i will try to debug later.